src/core/server/saved_objects/migrations/core/transform_saved_object_document_error.ts

```diff
@@ -10,9 +10,13 @@
  * Error thrown when saved object migrations encounter a transformation error.
  * Transformation errors happen when a transform function throws an error for an unsanitized saved object
  */
-
 export class TransformSavedObjectDocumentError extends Error {
-  constructor(public readonly originalError: Error) {
-    super(`${originalError.message}`);
+  constructor(public readonly originalError: Error, public readonly version: string) {
+    super(`Migration function for version ${version} threw an error`);
+    appendCauseStack(this, originalError);
   }
 }
+
+const appendCauseStack = (error: Error, cause: Error) => {
+  error.stack = (error.stack ?? '') + `\nCaused by:\n${cause.stack ?? cause.message}`;
+};
```

Comment:
So, I choose to append the stack trace to the `transform` error instead of just replacing the top error's stack with the cause's stack. It felt like this made the most sense, as that way we still have the info about both errors, in case of. 

(Also this is how languages properly handling nested errors are usually doing it, at least regarding the output)